scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/UploadServlet.java

```diff
@@ -149,76 +161,98 @@ private void fetchFromCatalogAndWriteResponse(String bucketName, String workflow
             HttpServletResponse response) {
         String encodedWorkflowName;
         try {
-            encodedWorkflowName = URLEncoder.encode(workflowName, "UTF-8");
+            encodedWorkflowName = URLEncoder.encode(workflowName, StandardCharsets.UTF_8.name());
         } catch (UnsupportedEncodingException e) {
             encodedWorkflowName = workflowName;
         }
         String url = URL_CATALOG + "/buckets/" + bucketName + "/resources/" + encodedWorkflowName + "/raw";
-        LOGGER.info("Sending request to catalog: " + url);
-        CloseableHttpClient httpclient = HttpClients.createDefault();
+        LOGGER.info("Sending request to catalog: {}", url);
+
         HttpGet httpget = new HttpGet(url);
-        httpget.addHeader("sessionID", sessionId);
-        CloseableHttpResponse httpResponse = null;
-        try {
-            httpResponse = httpclient.execute(httpget);
+        httpget.addHeader(PARAMS_SESSION_ID, sessionId);
+
+        try (CloseableHttpClient httpclient = HttpClients.createDefault();
+                CloseableHttpResponse httpResponse = httpclient.execute(httpget);) {
             HttpEntity responseBody = httpResponse.getEntity();
             File job = File.createTempFile("job_upload", ".xml");
             FileUtils.copyInputStreamToFile(responseBody.getContent(), job);
-            LOGGER.info(job.toString());
+            LOGGER.info("job file={}", job);
             writeResponse(job, response);
         } catch (Exception e) {
             LOGGER.error("Error when sending the request to catalog", e);
-        } finally {
-            try {
-                if (httpResponse != null) {
-                    httpResponse.close();
-                }
+        }
+    }
 
-            } catch (IOException e) {
-                LOGGER.error("Error when closing the response", e);
-            }
+    private void fetchJobXMLAndWriteResponse(HttpServletRequest request, HttpServletResponse response) {
+        String jobId = request.getHeader(HEADER_JOB_ID);
+        String sessionId = request.getHeader(PARAMS_SESSION_ID);
+
+        try (InputStream is = new ByteArrayInputStream(((SchedulerServiceImpl) Service.get()).getJobXML(sessionId,
+                                                                                                        jobId)
+                                                                                             .getBytes(StandardCharsets.UTF_8))) {
+            LOGGER.info("Fetching Job XML for job {}", jobId);
+            File job = File.createTempFile("job_resubmit", ".xml");
```

Comment:
To me, there is a big potential concurrent issue here. If all the re-submitted job files are going to be uploaded under the same name, there can be an overlapping of writes in this file, since many users can use the resubmit at the same time. If the job ID is still unique even for job resubmission (I hope so, because otherwise it would break a big part of the model), the you can append the job ID to the name of the file to make sure it is unique.

Reply:
@jrochas in fact, File.createTempFile will create a unique file name (uuid-like) surrounded by the given prefix & suffix so every time you call the servlet it creates a unique temp file then deletes it to free space.